azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java

```diff
@@ -207,12 +207,49 @@ public void canCreateLowPriorityVirtualMachine() throws Exception {
         Assert.assertEquals((Double) 1000.0, foundVM.billingProfile().maxPrice());
         Assert.assertEquals(VirtualMachineEvictionPolicyTypes.DEALLOCATE, foundVM.evictionPolicy());
 
+        // change max price
+        try {
+            foundVM.update()
+                    .withMaxPrice(1500.0)
+                    .apply();
+            // not run to assert
+            Assert.assertEquals((Double) 1500.0, foundVM.billingProfile().maxPrice());
+            Assert.assertTrue(false);
+        } catch (AssertionError e) {
+            throw e;
+        } catch (Exception e) {} // cannot change max price when vm is running
```

Comment:
Maybe better to catch CloudException (and then there would be no need to catch-rethrow AssertionError)? No need to record again.